### PR TITLE
Update AsynchronousReadIndirectBufferFromRemoteFS.cpp

### PR DESCRIPTION
**Pull request by kssenii**
### Changelog category (leave one):
- Not for changelog (changelog entry is not required)

### CODE REVIEW

**Comment by kssenii** [comment]
:(
```
Exception: 403 Client Error: rate limit exceeded for url: https://api.github.com/users/kssenii/orgs
```

**Review by alexey-milovidov — APPROVED** [review]
What is the effect of this change?
Is it possible to make a test?

File: src/Disks/IO/AsynchronousReadIndirectBufferFromRemoteFS.cpp [hunk 1]
```diff
@@ -133,14 +133,15 @@ void AsynchronousReadIndirectBufferFromRemoteFS::prefetch()
 
 void AsynchronousReadIndirectBufferFromRemoteFS::setReadUntilPosition(size_t position)
 {
-    if (prefetch_future.valid())
+    if (!read_until_position || position > *read_until_position)
```
Comment (alexey-milovidov) [inline src/Disks/IO/AsynchronousReadIndirectBufferFromRemoteFS.cpp:136]: Let's write a comment.

**Comment by kssenii** [comment]
> What is the effect of this change?
Is it possible to make a test?

The effect of the changes is explained in the comment that I just added (this PR removes resetting the prefect when we actually can avoid doing that). 

I was just carefully rereading this code and found this place, which could be improved. 

Not sure if the test can be added, this is too tricky place...

**Comment by alexey-milovidov** [comment]
Ok.

**Comment by kssenii** [comment]
> 00725_memory_tracking

🤔

**Comment by kssenii** [comment]
> AST fuzzer (debug) — Received signal 11

https://github.com/ClickHouse/ClickHouse/issues/43745

**Pull request merged**